lively-lions/MUD/dungeon/views/character_view.py

```diff
@@ -0,0 +1,184 @@
+                            character_name_list.append(obj.name)
+                    return HttpResponse(",".join(character_name_list))
+                else:
+                    return HttpResponse("invalid")
+            elif request.POST["view_name"] == 'attack_character':
+                # attack # mycharacter -> others
+                # parameter : target_user
+                # return True : Success attack {target_user} -> Dead | hp : {number}
+                # return Fail : return HttpResponse("invalid")
+                user = MudUser.objects.get(pk=request.user.pk)
+                if user.now_connected_character_name != '':
+                    now_character = Character.objects.get(name=user.now_connected_character_name)
+                    location_x = now_character.location_x
+                    location_y = now_character.location_y
+                    location_z = now_character.location_z
+                    character_name_list = []
+                    for obj in Character.objects.filter(location_x=location_x,
+                                                        location_y=location_y,
+                                                        location_z=location_z):
+                        if obj.user.now_connected_character_name == obj.name:
+                            character_name_list.append(obj.name)
+                    # pytest.set_trace()
+                    if request.POST["target_user"] in character_name_list and \
+                            user.now_connected_character_name != request.POST["target_user"]:
+                        attacker = Character.objects.get(name=user.now_connected_character_name)
+                        target = Character.objects.get(name=request.POST["target_user"])
+                        # Damage Formula : attack*(100/(100+defense))
+                        target.hp = target.hp - attacker.total_attack*(100/(100+target.total_defense))
+                        if target.hp <= 0:
+                            target.hp = target.max_hp
+                            target.location_x = 10
+                            target.location_y = 10
+                            target.location_z = 10
+                            target.save()
+                            message = "Success attack "+request.POST["target_user"] + " -> Dead"
+                            return HttpResponse(message)
+                        target.save()
+                        target_hp = Character.objects.get(name=request.POST["target_user"]).hp
+                        message = "Success attack "+request.POST["target_user"] + " -> hp : " + str(target_hp)
+                        return HttpResponse(message)
+                    else:
+                        return HttpResponse("invalid")
+                else:
+                    return HttpResponse("invalid")
+            else:
+                return HttpResponse("invalid")
+        else:
+            return HttpResponse("invalid")
```

Comment:
It's a very "fat" view. But it has been factored out into its own view, there are inline comments to explain whats going on, making it easy to follow along. The use of clear assignments, like how names are assigned right before they are used is great to see. This is overall skillfully written.